tests/scanner/scanners/external_project_access_scanner_test.py

```diff
@@ -0,0 +1,301 @@
+                                                    self.model_name,
+                                                    self.snapshot_timestamp,
+                                                    self.rules)
+        flattened_iter = scanner._flatten_violations([violation1])
+
+        flat_violation = flattened_iter.next()
+        self.assertEqual(flat_violation['resource_id'], '12345')
+        self.assertEqual(
+            flat_violation['resource_type'], resource_mod.ResourceType.PROJECT)
+        self.assertEqual(flat_violation['rule_name'], 'Only my org')
+        self.assertEqual(flat_violation['full_name'], 'projects/12345')
+        self.assertEqual(
+            flat_violation['violation_data']['member'], 'user1@example.com')
+
+        with self.assertRaises(StopIteration):
+            flat_violation = flattened_iter.next()
+
+    def tearDown(self):
+        epas.get_user_emails = self.stash_email_method
+
+
+class GetUserEmailsTest(ForsetiTestCase):
+    """Test the storage_helpers modue."""
+    def setUp(self):
+        self.engine = create_test_engine()
+        _session_maker = sessionmaker()
+        self.session = _session_maker(bind=self.engine)
+        initialize(self.engine)
+        res_user1 = ResourceMock('1', {'primaryEmail': 'user1@example.com',
+                                       'suspended': False},
+                                 'gsuite_user',
+                                 'resource')
+        res_user2 = ResourceMock('2', {'primaryEmail': 'user2@example.com',
+                                       'suspended': False},
+                                 'gsuite_user',
+                                 'resource')
+        res_user3 = ResourceMock('3', {'primaryEmail': 'user3@example.com',
+                                       'suspended': False},
+                                 'gsuite_user',
+                                 'resource')
+        self.resources = [res_user1, res_user2, res_user3]
+        self.storage = Storage(self.session)
+        _ = self.storage.open()
+        for resource in self.resources:
+            self.storage.write(resource)
+        self.storage.commit()
+        self.service_config = TestServiceConfig()
+
+    #pylint: disable=C0301,W9016,W9015,W0613
+    @mock.patch('google.cloud.forseti.scanner.scanners.external_project_access_scanner._get_inventory_storage')
+    @mock.patch('google.cloud.forseti.services.inventory.storage.DataAccess.get_latest_inventory_index_id')
+    def test_get_emails(self, mock_get_latest_inv_ndx_id, mock_storage):
+        """Test retrieving e-mails from storage"""
+
+        expected_emails = [u'user1@example.com',
+                           u'user2@example.com',
+                           u'user3@example.com']
+        mock_storage.return_value = self.storage
+        emails = epas.get_user_emails(self.service_config)
+        self.assertListEqual(emails, expected_emails)
+
+
+class OtherModuleTest(ForsetiTestCase):
```

Comment:
Do we really need an extra `OtheModuleTest` class?  Can we can put the `test_get_project_ids()` in `ExternalProjectAccessScannerTest` module?  See my comment below about your second test.

Reply:
moved to `ExternalProjectAccessScannerTest`